File: game/scripts/vscripts/items/item_death_orb/item_death_orb.lua
item_death_orb = class({})

function item_death_orb:OnSpellStart()
    local charges = self:GetCurrentCharges()
    local caster = self:GetCaster()
    local heal = 24

    CustomEntitiesLegacy:GiveManaAndEnergy(caster, charges, true, true)

    if charges == 100 then
        CustomEntitiesLegacy:TrueHeal(caster, heal)
    end
end

File: src/vscripts/items/item_death_orb.ts
import { CustomItem } from "../abilities/framework/custom_ability";
import { registerAbility } from "../lib/dota_ts_adapter";
import { giveManaAndEnergy } from "../util";

@registerAbility("item_death_orb")
class ItemDeathOrb extends CustomItem {
    OnSpellStart() {
        const charges = this.GetCurrentCharges();
        const heal = 24;

        giveManaAndEnergy(this.GetCaster(), charges, true, true);

        if (charges == 100) {
            CustomEntitiesLegacy.TrueHeal(this.GetCaster(), heal);
        }
    }
}
